src/gt4py/definitions.py

```diff
@@ -101,6 +101,7 @@ class BuildOptions(AttributeClassLike):
     backend_opts = attribute(of=DictOf[str, Any], factory=dict)
     build_info = attribute(of=dict, optional=True)
     rebuild = attribute(of=bool, default=False)
+    raise_if_not_cached = attribute(of=bool, default=False)
```

Comment:
Is the use of `rebuild` and `raise_if_not_cached` flags clear to the user? Would a flag `rebuild` with the settings `force`, `disable` and `default` not make more sense. What happens if `rebuild = True` and `raise_if_not_cached = True`?